src/TypeMirror.spec.lua

```diff
@@ -0,0 +1,53 @@
+return function()
+	local Type = require(script.Parent.Type)
+	local Mirror = require(script.Parent.TypeMirror)
+
+	describe("Type", function()
+		it("should return a mirror of an internal type", function()
+			local name = Type.nameOf(Type.Element)
+			local mirroredType = Mirror.Type[name]
+			expect(mirroredType).to.equal(Mirror.Type.Element)
+		end)
+
+		it("should not return the actual internal type", function()
+			local name = Type.nameOf(Type.Element)
+			local mirroredType = Mirror.Type[name]
+			expect(mirroredType).to.never.equal(Type.Element)
+		end)
+
+		it("should include all allowed types", function()
+			for _, type in ipairs(Mirror.typeList) do
+				local name = Type.nameOf(type)
+				local mirroredType = Mirror.Type[name]
+				expect(mirroredType).to.be.ok()
+			end
+		end)
+
+		it("should not include any other types", function()
+			local name = Type.nameOf(Type.VirtualNode)
+			local success = pcall(function()
+				local _ = Mirror.Type[name]
+			end)
+			expect(success).to.equal(false)
+		end)
+	end)
+
+	describe("typeOf", function()
+		it("should throw if the value is not a valid type", function()
+			expect(pcall(Mirror.typeOf, 1)).to.equal(false)
+			expect(pcall(Mirror.typeOf, true)).to.equal(false)
+			expect(pcall(Mirror.typeOf, "test")).to.equal(false)
+			expect(pcall(Mirror.typeOf, print)).to.equal(false)
+			expect(pcall(Mirror.typeOf, {})).to.equal(false)
+			expect(pcall(Mirror.typeOf, newproxy(true))).to.equal(false)
+		end)
```

Comment:
Should these throw? Looking at the code it doesn't seem like they should, and I don't think they should from a type perspective. We don't know if something is a Roact object until we call `typeOf` on it!

Reply:
It throws because none of these are valid keys when getting the `nameOf` the real type, and it seemed reasonable to throw if you're not passing in a plausible symbol (userdata), but I think not throwing and returning nil sounds good too. Updated.